File: gpt-3.5-ft-large/stage2/loop_construct_seq_clause.f90
#ifndef T1
!T1:loop construct seq clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER,DIMENSION(LOOPCOUNT):: a, a_copy
        REAL(2.0, 4.0)::coeff
        REAL*8, DIMENSION(5*LOOPCOUNT)::random_number
        REAL*8:: sum = 0
        INTEGER :: x
        SEEDDIM(1)=1
        SEED(1)=627
        
        CALL SRAND48(627)
        
        a_copy = a
        DO x = 1, 5 * LOOPCOUNT
          CALL DRAND48(random_number(x))
        END DO
        DO x = 1, LOOPCOUNT
          a(x) = 5
        END DO
        
        !$acc data copy(a(1:LOOPCOUNT))
          !$acc serial
            !$acc loop seq
            DO x = 1, LOOPCOUNT
              a(x) = 1
            END DO
          !$acc end serial
        !$acc end data
        DO x = 1, LOOPCOUNT
          sum = sum + (a(x) - 1)
        END DO
        DO x = 1, LOOPCOUNT
          sum = sum + a(x)
        END DO
        DO x = 1, 5 * LOOPCOUNT
          sum = sum + (random_number(x) - a(x/5))
        END DO
        IF (sum .neq. 0) THEN
          errors = errors + 1
        END IF

        a = a_copy
        IF (errors .lt. 1) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: gpt-3.5-finetuned/stage2/loop_construct_seq_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
//T1:loop construct seq clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    for (int x = 0; x < n; ++x){
        b[x] = 1.5;
    }

    #pragma acc loop seq
    for (int x = 0; x < n; ++x){
        b[x] = b[x] * 2;
    }

    for (int x = 0; x < n; ++x){
        if (fabs(b[x] - 3) > PRECISION){
            err = 1;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}